### PR TITLE
Profile cache invalidation when linked geography hierarchies are updated #139

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
Invalidate cache for profile when linked geography hierarchy is updated.

## Related Issue
#139 

## How to test it locally
save geography hierarchy from admin panel and api should load hierarchy changes.
I checked if dev server logs updated.

## Changelog
* Added invalidate cache signal for geography hierarchy in cache.py
* Added test for change above

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out

### Commits

- [x]  commits are clean
- [x]  commit messages are clean

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers
- [x]  black was run locally (as part of the pre-commit hook)

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

**Review by milafrerichs — CHANGES_REQUESTED** [review]
can you add a signal for when the geography changes? just for completeness 
thanks.

**Comment by goyal1092** [comment]
@milafrerichs I have updated the PR.
Please review

File: tests/test_cache.py [hunk 1]
```diff
@@ -168,9 +171,19 @@ def test_invalidate_profile_cache_for_geography_udpate(self, mock_update_profile
...
+        profile1 = ProfileFactory(geography_hierarchy=geography_hierarchy)
+        profile2 = ProfileFactory(geography_hierarchy=geography_hierarchy)
+        indicator_data = IndicatorDataFactory(geography=geography)
+
+        self.assertEqual(mock_update_profile_cache.call_count, 4)
         calls = [
             call(datasetdata.dataset.profile),
+            call(profile1),
+            call(profile2),
+            call(indicator_data.indicator.dataset.profile),tests/test_cache.py
```
Comment (milafrerichs) [inline tests/test_cache.py:186]: there is something leftover. :)

**Review by milafrerichs — CHANGES_REQUESTED** [review]

File: tests/test_cache.py (hunk 1 above)
Comment (goyal1092) [reply tests/test_cache.py]: 😱 🙀

Comment (goyal1092) [reply tests/test_cache.py]: Fixed @milafrerichs

File: wazimap_ng/cache.py [hunk 2]
```diff
@@ -148,11 +157,32 @@ def indicator_updated(sender, instance, **kwargs):
...
+    # indicator data objs
+    indicator_data_profile_ids = list(instance.indicatordata_set.values_list(
+        "indicator__dataset__profile", flat=True
+    ).order_by("indicator__dataset__profile").distinct())
+
+    # Get hierarchy profile linked to 
+    hierarchy_profile_ids = list(instance.geographyhierarchy_set.values_list(
+        "profile", flat=True
+    ).order_by("profile").distinct())
+    print(hierarchy_profile_ids)
```
Comment (milafrerichs) [inline wazimap_ng/cache.py:174]: leftover print statement

**Review by milafrerichs — CHANGES_REQUESTED** [review]

File: wazimap_ng/cache.py (hunk 2 above)
Comment (goyal1092) [reply wazimap_ng/cache.py]: Fixed @milafrerichs

Comment (goyal1092) [reply wazimap_ng/cache.py]: thanks

**Review by milafrerichs — APPROVED** [review]
